pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -458,21 +458,14 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(node *model.Pr
 				}
 				gatewayRoutes[gatewayName][vskey] = routes
 			}
-			// This is the service that is exposed on gateway using VirtualService.
-			var gatewayService *model.Service
-			for _, hostname := range intersectingHosts {
-				if svc, exists := nameToServiceMap[hostname]; exists {
-					gatewayService = svc
-				}
-			}
-
 			for _, hostname := range intersectingHosts {
 				if vHost, exists := vHostDedupMap[hostname]; exists {
 					vHost.Routes = append(vHost.Routes, routes...)
 					if server.Tls != nil && server.Tls.HttpsRedirect {
 						vHost.RequireTls = route.VirtualHost_ALL
 					}
 				} else {
+					gatewayService := nameToServiceMap[hostname]
```

Comment:
This seems like a behavioral change, but no tests were changed. Can you add tests that cover these cases?

Reply:
Yes. Will add. I realized there are no tests for gateway